modules/spring-config/src/main/java/org/mule/runtime/config/internal/factories/EnrichedErrorMappingsFactoryBean.java

```diff
@@ -45,15 +44,13 @@ public class EnrichedErrorMappingsFactoryBean extends AbstractComponentFactory<E
 
   @Override
   public EnrichedErrorMapping doGetObject() throws Exception {
-    ComponentIdentifier sourceType = source != null ? buildFromStringRepresentation(source) : ANY;
-
-    ErrorType errorType = errorTypeRepository
-        .lookupErrorType(sourceType)
-        .orElseThrow(() -> new MuleRuntimeException(createStaticMessage("Could not find error '%s'.",
-                                                                        source)));
-
-    ErrorTypeMatcher errorTypeMatcher =
-        new SingleErrorTypeMatcher(errorType);
+    ErrorTypeMatcher errorTypeMatcher;
+    try {
+      errorTypeMatcher = source != null ? createErrorTypeMatcher(errorTypeRepository, source)
+          : createErrorTypeMatcher(errorTypeRepository.lookupErrorType(ANY).get());
+    } catch (IllegalArgumentException e) {
+      throw new MuleRuntimeException(createStaticMessage("Could not find error '%s'.", source));
```

Comment:
for the case with comma separated error this would show all the errors, not the one in particualr that failed. You should jsut use the message from the caught exception